DataStructures/Queues/Deque.java

```diff
@@ -0,0 +1,163 @@
+/**Implementation of double ended queue*/
+
+import java.util.Scanner;
+
+public class Deque {
+	private int maxSize;
+	private int f, r, a[];
```

Comment:
Variables name should reflect what it is not , instead of using f , r ,a you should use first , rear , queue or something of your choice but descriptive in nature

Reply:
Added descriptive names